src/NUnitTestAdapter/NUnit/AssemblyHelper.cs

```diff
@@ -0,0 +1,174 @@
+// ***********************************************************************
+// Copyright (c) 2008 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.IO;
+using System.Reflection;
+
+namespace NUnit.Engine.Internal
+{
+    /// <summary>
+    /// AssemblyHelper provides static methods for working
+    /// with assemblies.
+    /// </summary>
+    public class AssemblyHelper
+    {
+        #region GetAssemblyPath
+
+#if !SILVERLIGHT && !PORTABLE
+        /// <summary>
+        /// Gets the path from which the assembly defining a type was loaded.
+        /// </summary>
+        /// <param name="type">The Type.</param>
+        /// <returns>The path.</returns>
+        public static string GetAssemblyPath(Type type)
```

Comment:
Since we are not constrained by .NET 2.0 here, all of these helper methods could be converted to extension methods for cleaner code.